src/testkit/scala/tools/testkit/AssertUtil.scala

```diff
@@ -92,6 +92,9 @@ object AssertUtil {
       fail("Expression did not throw!")
     } catch {
       case e: T if checkMessage(e.getMessage) =>
+      case t: Throwable                       =>
+        val clazz = scala.reflect.classTag[T].runtimeClass
+        fail(s"wrong exception: expected ${clazz.getName} but was ${t.getClass.getName}")
```

Comment:
This has a slightly better formulation at L112 or so in `assertThrown`.

I think `assertThrows` could be written in terms of `assertThrown`.

Reply:
That is, I like that it uses `NonFatal` and suppressed exception. The test is called `discriminating`, enough said.

Reply:
So it does.  Let me see if it can.